gradio/components/image.py

```diff
@@ -276,6 +282,10 @@ def preprocess(
 
         if self.tool == "sketch" and self.source in ["upload", "webcam"]:
             mask_im = processing_utils.decode_base64_to_image(mask)
+
+            if mask_im.mode == "RGBA":  # whiten any opaque pixels in the mask
```

Comment:
I don't follow this. Can you explain what's happening?

Reply:
yep, so previously in the frontend, whenever a user drew a mask, we were drawing it twice - once on the visible layer directly on the image, and then again on a hidden layer called "mask", which instead drew on a white brush on a black background. This mask layer was what was getting sent to the backend on submit.
I simplified the frontend so that there is a single mask layer, now visible, that hovers over the drawing. A user is no longer drawing directly on the image, but instead on this visible mask layer. On submit, we send this mask layer. However, this mask layer is no longer the white on black image of before - it has opacity and may be whatever color that brush_color has been set to. So now we convert this mask layer into the B+W mask of before in the backend with these lines of code.
We support "backwards compatibility" in case someone uses a client, with `if mask_im.mode == "RGBA": `, because previously masks were only RGB.
cc @pngwn for context to the frontend changes

Reply:
Are we certain that this conversion is always accurate (going from colours to b+w, especially since the sketches are antialiased)? Masks are typically alpha layers and lack any colours, I'm curious about why we would ever want a mask layer with colour.

Reply:
Confirmed that antialiasing still works - there are values between 0 and 255 at the edges of the mask